src/ci/docker/x86_64-gnu-aux/Dockerfile

```diff
@@ -16,10 +16,61 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
   libgl1-mesa-dev \
   llvm-dev \
   libfreetype6-dev \
-  libexpat1-dev
+  libexpat1-dev \
+  gnupg \
+  apt-utils \
+  wget
+
+RUN curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
+RUN apt install -y nodejs
+
+# This part is to install the requirements to run rustdoc GUI tests
+#
+# To do so we need:
+#  * a chrome instance (we can extend it to firefox)
+#  * cloning the browser-UI-test repository from https://github.com/GuillaumeGomez/browser-UI-test/
+#  * cloning the test-rust-docs-ui repository from https://github.com/GuillaumeGomez/test-rust-docs-ui
+#
+# The browser-UI-test repository is a framework to make it as easy as possible to write GUI tests
+# where the test-rust-docs-ui repository contains the tests specific to rustdoc GUI.
+#
+# To prevent unwanted changes breaking tests for everyone, both repositories will target a specfic
+# commit that will need to be updated when the rustdoc GUI will be updated.
+RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  google-chrome-unstable \
+  fonts-ipafont-gothic \
+  fonts-wqy-zenhei \
+  fonts-thai-tlwg \
+  fonts-kacst \
+  fonts-freefont-ttf \
+  && rm -rf /var/lib/apt/lists/*
+
+RUN git clone https://github.com/GuillaumeGomez/browser-UI-test/ \
+  && cd browser-UI-test \
+  && git checkout 1de38b0962f9abbddc677aaa3a94b864c0f40668 \
+  && rm -rf tests \
+  && npm install
+RUN git clone https://github.com/GuillaumeGomez/test-rust-docs-ui/ \
+  && cd test-rust-docs-ui \
+  && git checkout 03ea8284c2ab62607b6260793f08ea34953b14be
+
 
 COPY scripts/sccache.sh /scripts/
 RUN sh /scripts/sccache.sh
 
 ENV RUST_CONFIGURE_ARGS --build=x86_64-unknown-linux-gnu
 ENV RUST_CHECK_TARGET check-aux
+
+# Now we run the rustdoc GUI tests and hope it all works out
+RUN cd test-rust-docs-ui/test-docs && rustdoc src/lib.rs
+
+RUN cd test-rust-docs-ui \
+  && node ../browser-UI-test/src/index.js --no-sandbox \
+    --test-folder ui-tests \
+    --failure-folder failures/ \
+    --variable DOC_PATH test-docs/doc/test_docs \
+    --show-text \
+    --generate-images
```

Comment:
Can we move the git clones and this bit to bootstrap? That way we have this better integrated into the normal testing workflow, even if it can only reliably run from this docker container (due to subtle differences in rendering).

Can you create a step in src/bootstrap/test.rs that clone the github repositories and runs node? I can help integrate that into the check-aux target once that's done.

Reply:
Sure!